File: src/Client.Desktop/View/TextureHolder.cs
using Microsoft.Xna.Framework;
using Microsoft.Xna.Framework.Graphics;

namespace Client.View
{
    /// <summary>
    /// TextureHolder represents part of a texture. Because of Monogame/XNA limitations, can' create texture
    /// in unit tests, so TextureHolder is natural replacement of Texture2D.
    /// THe calss can be extended for purpose of Dependency Injection to mark individual types types of textured
    /// </summary>
    public class TextureHolder
    {
        /// <summary>
        /// Parameterless constructor used in tests to allow create an instance and make assertion.
        /// </summary>
        public TextureHolder() { }

        public readonly Texture2D Texture2D;
        public readonly Rectangle Source;
        public TextureHolder(Texture2D texture, Rectangle source)
        {
            this.Texture2D = texture;
            this.Source = source;
        }
    }
}

File: core/src/com/mygdx/game/view/TextureHolder.java
package com.mygdx.game.view;

import com.badlogic.gdx.graphics.Texture;

/**
 *  TextureHolder represents part of a texture. Because of game engine limitation, can' create texture
 *  in unit tests, so TextureHolder is a natural replacement of Texture.
 *  THe calss can be extended for purpose of Dependency Injection to mark individual types types of textured
 */

public class TextureHolder {
    /**
     * Parameterless constructor used in tests to allow create an instance and make assertion.
     */
    public TextureHolder() {
        this(null, null);
    }

    public final Texture texture;
    public final IntRectangle source;

    public TextureHolder(Texture texture, IntRectangle source)
    {
        this.texture = texture;
        this.source = source;
    }
}
